sphinx/screens-detail/subscription/subscription/src/main/java/chat/sphinx/subscription/ui/SubscriptionViewModel.kt

```diff
@@ -0,0 +1,246 @@
+                            if (subscription == null) {
+                                navigator.popBackStack()
+                            } else {
+                                when(subscriptionRepository.deleteSubscription(subscription.id)) {
+                                    is Response.Error -> {
+                                        submitSideEffect(
+                                            SubscriptionSideEffect.Notify(app.getString(R.string.failed_to_delete_subscription))
+                                        )
+                                    }
+                                    is Response.Success -> {
+                                        navigator.popBackStack()
+                                    }
+                                }
+                            }
+                        }
+                    }
+                }
+            )
+        }
+    }
+
+    fun pauseSubscription() {
+        viewModelScope.launch(mainImmediate) {
+            subscriptionRepository.getActiveSubscriptionByContactId(ContactId(args.argContactId)).firstOrNull().let { subscription ->
+                if (subscription == null) {
+                    submitSideEffect(
+                        SubscriptionSideEffect.Notify(app.getString(R.string.failed_to_pause_subscription))
+                    )
+                } else {
+                    when (subscriptionRepository.pauseSubscription(subscription.id)) {
+                        is Response.Error -> {
+                            submitSideEffect(
+                                SubscriptionSideEffect.Notify(app.getString(R.string.failed_to_pause_subscription))
+                            )
+                        }
+                        is Response.Success -> {
+                            submitSideEffect(
+                                SubscriptionSideEffect.Notify(app.getString(R.string.successfully_paused_subscription))
+                            )
+                            updateViewState(
+                                SubscriptionViewState.SubscriptionLoaded(
+                                    Subscription(
+                                        id = subscription.id,
+                                        subscription.cron,
+                                        subscription.amount,
+                                        subscription.end_number,
+                                        subscription.count,
+                                        subscription.end_date,
+                                        subscription.ended,
+                                        paused = true,
+                                        subscription.created_at,
+                                        subscription.updated_at,
+                                        subscription.chat_id,
+                                        subscription.contact_id
+                                    )
```

Comment:
unnecessary. Remove.

success case returns a paused, updated subscription object and can be set w/o need to re-instantiate a `Subscription` object.